go.mod

```diff
@@ -9,30 +11,32 @@ require (
 	cloud.google.com/go/resourcemanager v1.10.2
 	github.com/GoogleCloudPlatform/k8s-cloud-provider v1.33.0
 	github.com/go-logr/logr v1.4.2
-	github.com/google/go-cmp v0.6.0
-	github.com/googleapis/gax-go/v2 v2.14.0
-	github.com/onsi/ginkgo/v2 v2.22.2
-	github.com/onsi/gomega v1.36.2
+	github.com/google/go-cmp v0.7.0
+	github.com/googleapis/gax-go/v2 v2.14.1
+	github.com/onsi/ginkgo/v2 v2.23.3
+	github.com/onsi/gomega v1.36.3
 	github.com/pkg/errors v0.9.1
-	github.com/spf13/pflag v1.0.5
+	github.com/spf13/pflag v1.0.6
 	github.com/stretchr/testify v1.10.0
-	golang.org/x/crypto v0.32.0
-	golang.org/x/mod v0.22.0
-	golang.org/x/net v0.34.0
-	google.golang.org/api v0.214.0
+	golang.org/x/crypto v0.36.0
+	golang.org/x/mod v0.23.0
+	golang.org/x/net v0.37.0
+	google.golang.org/api v0.215.0
 	google.golang.org/grpc v1.67.3
-	k8s.io/api v0.31.3
-	k8s.io/apimachinery v0.31.3
-	k8s.io/client-go v0.31.3
-	k8s.io/component-base v0.31.3
+	k8s.io/api v0.32.3
+	k8s.io/apimachinery v0.32.3
+	k8s.io/client-go v0.32.3
+	k8s.io/component-base v0.32.3
 	k8s.io/klog/v2 v2.130.1
-	k8s.io/utils v0.0.0-20240711033017-18e509b52bc8
-	sigs.k8s.io/cluster-api v1.9.4
-	sigs.k8s.io/cluster-api/test v1.9.4
-	sigs.k8s.io/controller-runtime v0.19.4
+	k8s.io/utils v0.0.0-20241104100929-3ea5e8cea738
+	sigs.k8s.io/cluster-api v1.10.0-beta.0.0.20250325200233-8523bb29cf5a
```

Comment:
As this is a CAPI bump could you change the PR title and also ensure the CAPI versions and Go versions are updated throughout the project. For example, in the e2e test config files.

Reply:
I have updated the pr title. I don't know what should be changed "in the e2e test config files".

Reply:
I think you missed this one @tamalsaha

Reply:
Hello, I have updated it now. I don't understand what is happening here. So, please check thoroughly.

https://github.com/kubernetes-sigs/cluster-api-provider-gcp/pull/1459/commits/b9c2994c3e3576927bed670df96525ada4def363

Reply:
How do you tell what etcd version is there?

Reply:
I normally look at the version used for the same core CAPI version